### PR TITLE
Fix issue 3364

### LINKED ISSUE #3364: [FEATURE] Re-design "Categories" view
### Please confirm the following:

- [X] I've checked [the current issues](https://github.com/Ivy-Apps/ivy-wallet/issues) for duplicate issues.
- [X] I've requested a single (only one) feature/change in this issue. It complies with the One Request Per GitHub Issue (ORPGI) rule.
- [X] My issue is well-defined and describes how it should be implemented from UI/UX perspective.

### What do you want to be added or improved?

_Note:_ Similar concern for different view has been raised here - #3297 

Most if not all, categories will have "expense" (not income) - Consider this screenshot. The space here is mostly wasted.

SS

### Why do you need it?

_No response_

### How do you imagine it?

_No response_

### PR DESCRIPTION
**Pull request by rodrigomatosc**
## Pull request (PR) checklist

Please check if your pull request fulfills the following requirements:

- [x] I've read the [Contribution Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/CONTRIBUTING.md) and my PR doesn't break the rules.
- [x] I've read and understand the [Developer Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/Guidelines.md).
- [x] I confirm that I've run the code locally and everything works as expected.
- [x] My PR includes only the necessary changes to fix the issue (i.e., no unnecessary files or lines of code are changed).
- [x] 🎬 I've attached a **screen recording** of using the new code to the next paragraph (if applicable).

## What's changed?

Describe with a few bullets **what's new:**
- I've change the  design of Categories screen.

Before|After
---|---
image

image

## Risk factors

**What may go wrong if we merge your PR?**
 - No known issue.

**In what cases won't your code work?**
- No known issue.

## Does this PR close any GitHub issues? (do not delete)

- Closes #3364

## Troubleshooting GitHub Actions (CI) failures ❌
Pull request checks failing? Read our [CI Troubleshooting guide](https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/CI-Troubleshooting.md).

### CODE REVIEW

File: shared/domain/src/main/java/com/ivy/domain/features/IvyFeatures.kt [hunk 1]
```diff
@@ -5,21 +5,30 @@ import javax.inject.Singleton
...
     override val compactAccountsMode = BoolFeature(
         key = "compact_account_ui",
         name = "Compact account UI",
         description = "Enables more compact and dense UI for the \"Accounts\" tab"
     )
 
+    override val compactCategoriesMode = BoolFeature(
+        key = "compact_categories_ui",
+        name = "Compact category UI",
+        description = "Activates a more streamlined and space-efficient interface for the \"Categories\" tab"
```
Comment (ILIYANGERMANOV) [inline shared/domain/src/main/java/com/ivy/domain/features/IvyFeatures.kt:25]: "Categories" screen,  not tab

File: screen/categories/src/main/java/com/ivy/categories/CategoriesScreen.kt [hunk 2]
```diff
@@ -556,10 +660,17 @@ private fun SelectTypeButton(
 
 @Preview
 @Composable
-private fun Preview(theme: Theme = Theme.LIGHT) {
+private fun PreviewCategoriesCompactModeEnabled(theme: Theme = Theme.LIGHT) {
```
Comment (ILIYANGERMANOV) [inline screen/categories/src/main/java/com/ivy/categories/CategoriesScreen.kt:663]: You need to record Paparazzi screneshots for this module. See the CI guide at the bottom of your PR description

**Review by ILIYANGERMANOV — COMMENTED** [review]
LGTM! Left some comments and we ca  merge after they're fixed

**Review by ILIYANGERMANOV — CHANGES_REQUESTED** [review]
The UI on the Categories screen should display "+" or "-" before the amount depending on whether the net balance for the category is positive. 

when {
  x > 0 -> "+"
  0 -> ""
  x < 0 -> "-"
}

**Comment by ILIYANGERMANOV** [comment]
> The UI on the Categories screen should display "+" or "-" before the amount depending on whether the net balance for the category is positive.
> 
> when { x > 0 -> "+" 0 -> "" x < 0 -> "-" }

This might not be caused by your PR but we should fix it 🙏

**Comment by rodrigomatosc** [comment]
> The UI on the Categories screen should display "+" or "-" before the amount depending on whether the net balance for the category is positive.
> 
> when { x > 0 -> "+" 0 -> "" x < 0 -> "-" }

The current rule implementeg in the default card header is : 

image

Would you like me to modify the implementation to match the one you indicated ?

**Comment by ILIYANGERMANOV** [comment]
> > The UI on the Categories screen should display "+" or "-" before the amount depending on whether the net balance for the category is positive.
> > when { x > 0 -> "+" 0 -> "" x < 0 -> "-" }
> 
> The current rule implementeg in the default card header is :
> 
> image
> Would you like me to modify the implementation to match the one you indicated ?

Yup, with the new design not showing +/- makes it confusing. I read between the lines but end goal is the Category screen to makes sense for the user

File: screen/categories/src/test/snapshots/images/com.ivy.categories_CategoriesScreenPaparazziTest_snapshot Categories compact Screen[Dark].png
Comment (ILIYANGERMANOV) [inline screen/categories/src/test/snapshots/images/com.ivy.categories_CategoriesScreenPaparazziTest_snapshot Categories compact Screen[Dark].png:1]: Just remove the space between "+ 2,150" and we're good to merge 👍 Tag me for re-review and will merge it

**Review by ILIYANGERMANOV — APPROVED** [review]

**Pull request merged**